### ISSUE TITLE
sqlite error when discovering files

### ISSUE THREAD

**Issue by Dromantor** [microsoft/vscode-cpptools#9066]
I cranked up logging to 11 ("7") to see if I can find something related to my symlink issue #8823 when I discovered this:

```
[lots of other files and directories]
  Processing folder (recursive): D:/DOKUMENTE/ARBEIT/PROJEKTE/OFF-GRID POWER SYSTEM/BMS-PLATINE/BMS-FIRMWARE.X/SRC/
process_path: D:/DOKUMENTE/ARBEIT/PROJEKTE/OFF-GRID POWER SYSTEM/BMS-PLATINE/BMS-FIRMWARE.X/SRC/
update_file_if_needed: D:\DOKUMENTE\ARBEIT\PROJEKTE\OFF-GRID POWER SYSTEM\BMS-PLATINE\BMS-FIRMWARE.X\SRC\# SYMLINK 'FIRMWARE-SHARED' AS 'SHARED' HERE FOR SUCCESS
[rc=0x80004005] (code_store::a_statement::prepare) SQLite Error 1: SQL logic error: near "FIRMWARE": syntax error
SQL text: SELECT DISTINCT f.* FROM files AS f WHERE f.leaf_name= '# SYMLINK 'FIRMWARE-SHARED' AS 'SHARED' HERE FOR SUCCESS'
  File added. Time stamp: 0
  Excluding file from tag parse due to unknown extension.
[lots of other files and directories]
```

where `# SYMLINK 'FIRMWARE-SHARED' AS 'SHARED' HERE FOR SUCCESS` is a simple file with no extension/content I created as an instruction to whoever wants to build my code bc. symlinks aren't synced across devices by my sync app.

Looks a lot like an input sanitaization problem (`'` is valid in filenames).

**Comment by sean-mcmanus** [comment]
The error only appears to cause the file to be failed to be added to our database, so it's not found via a "file symbol search", which seems low priority. Let us know if you know of any higher priority bug this may cause.

**Comment by Dromantor** [comment]
I didn't find any further problems caused by this exept the error message I posted. Symbol search did also work for me (didn't test extensively). SQL injection tests (`asdf'; drop table files;`) didn't cause any problems, seems everything after the semicolon is ignored. Just reported it because I noticed it, so no priority for me.

**Comment by sean-mcmanus** [comment]
Yeah, that's for reporting this. Our database expert on the VS teams says it's a simple fix and the correct escaping is already done in another spot, but it might not make it in until 1.11.0.

**Comment by sean-mcmanus** [comment]
This appears to be fixed with 1.10.7.
